query-service-impl/src/test/java/org/hypertrace/core/query/service/attribubteexpression/AttributeExpressionNormalizationTransformationTest.java

```diff
@@ -0,0 +1,75 @@
+package org.hypertrace.core.query.service.attribubteexpression;
+
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createAliasedAttributeExpression;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createAliasedColumnExpression;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createAliasedFunctionExpression;
+import static org.hypertrace.core.query.service.QueryRequestBuilderUtils.createColumnExpression;
+import static org.hypertrace.core.query.service.QueryRequestUtil.createSimpleAttributeExpression;
+import static org.hypertrace.core.query.service.QueryRequestUtil.createStringLiteralValueExpression;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.hypertrace.core.query.service.QueryFunctionConstants;
+import org.hypertrace.core.query.service.QueryTransformation.QueryTransformationContext;
+import org.hypertrace.core.query.service.api.Filter;
+import org.hypertrace.core.query.service.api.Operator;
+import org.hypertrace.core.query.service.api.OrderByExpression;
+import org.hypertrace.core.query.service.api.QueryRequest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class AttributeExpressionNormalizationTransformationTest {
+
+  @Mock QueryTransformationContext mockTransformationContext;
+
+  private final AttributeExpressionNormalizationTransformation transformation =
+      new AttributeExpressionNormalizationTransformation();
+
+  @Test
+  void transformColumnInAllPositions() {
+    QueryRequest originalRequest =
+        QueryRequest.newBuilder()
+            .addSelection(createAliasedColumnExpression("select-col", "select-alias"))
+            .addAggregation(
+                createAliasedFunctionExpression(
+                    QueryFunctionConstants.QUERY_FUNCTION_SUM,
+                    "agg-alias",
+                    createColumnExpression("agg-col").build()))
+            .setFilter(
+                Filter.newBuilder()
+                    .setLhs(createAliasedColumnExpression("filter-col", "filter-alias"))
+                    .setOperator(Operator.EQ)
```

Comment:
This looks fine to me, but it seems to me that we might break `CONTAIN_KEY_VALUE` in transition.

Reply:
You mentioned this in the previous comment, but I think I may be missing how this would break. Could you be more specific?

Reply:
I was thinking that `string_array` part is broken as you described here - https://github.com/hypertrace/query-service/pull/129#discussion_r775325504 - if we translate as is. But, I relooked, and the method `convertExpressionToMapLiterals` in `QueryRequestToPinotSQLConverter` takes care of it for now.